### PR TITLE
Fix uploader types validation

### LINKED ISSUE #27237: An exception is thrown when adding a Menu thumbnails image in Sell > Catalog > Categories > edit
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

When I edit a category and I try to add a Menu thumbnails picture, an exception is thrown
Capture d’écran 2022-01-06 à 12 02 32

### Expected behavior

I should be able to upload a Menu thumbnails picture

### Steps to reproduce

1. Go to Sell > Catalog > Categories
2. Edit a category
3. Choose a picture for Menu thumbnails
4. Click save -> Exception is thrown

### PrestaShop version(s) where the bug happened

develop

### PHP version(s) where the bug happened

7.4

### If your bug is related to a module, specify its name and its version

_No response_

### LINKED ISSUE #27303: BO - An exception is thrown when fill all the fields of a new category
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

An exception is thrown when fill all the fields of a new category (adding images for each field Category cover image, Category thumbnail and Menu thumbnails in the same time)
image

develop :x: 
https://watch.screencastify.com/v/LtEnuXige4UbQoKBvxEn

PS1778 : OK
https://watch.screencastify.com/v/MHCHO88qRYwkVpnYIvoM

PS1782 : OK
https://watch.screencastify.com/v/QS5JNUrMJ34P6qfUMU6x

### Expected behavior

No exception is thrown.

### Steps to reproduce

1. Bo > SELL > Catalog > Categories
2. Click on `Add new categoy`
3. Fill all the fields with an image of each field `Category cover image, Category thumbnail and Menu thumbnails`
4. Click on Save
5. See error >> An exception is generated

### PrestaShop version(s) where the bug happened

develop

### PHP version(s) where the bug happened

7.3

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by atomiix**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Since PR #26164, the default value for `$_accept_types` is an empty array (it was `null` before) and the validation processes didn't take the empty array for "accept all" like it was when the default value was `null`. This PR fixes it by taking an empty array as "accept all" types.
| Type?             | bug fix
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #27237 & #27303
| How to test?      | Please see #27237 & #27303
| Possible impacts? | 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27240)

### CODE REVIEW

**Review by matthieu-rolland — DISMISSED** [review]

**Review by matks — DISMISSED** [review]

**Comment by marwachelly** [comment]
Hello,

I tested the PR and I noticed the following problems:

1. I cannot delete the image uploaded definitively while editing the category, this issue is a regression on the branch develop and it exists for all images in category. So I created this [issue](https://github.com/PrestaShop/PrestaShop/issues/27311 ) And it was possible to change the image. But only for Menu thumbnails it is not possible even to change the image and when trying more then 3 times an exception is displayed. 
https://watch.screencastify.com/v/uHUzQZ58vfdXYlRg7dVW
2. For Category cover image and Category thumbnail when attaching a PDF for exemple the following alert is displayed : 
image
For Menu thumbnails when attaching a PDF for exemple the following alert is displayed
image

Could you please check?

Many thanks,

File: classes/ImageManager.php [hunk 1]
```diff
@@ -476,7 +476,7 @@ public static function isRealImage($filename, $fileMimeType = null, $mimeTypeLis
     public static function isCorrectImageFileExt($filename, $authorizedExtensions = null)
     {
         // Filter on file extension
-        if ($authorizedExtensions === null) {
+        if ($authorizedExtensions === null || empty($authorizedExtensions)) {
```
Comment (saulaski) [inline classes/ImageManager.php]:
```suggestion
        if (empty($authorizedExtensions)) {
```

**Review by saulaski — DISMISSED** [review]

File: classes/Uploader.php [hunk 2]
```diff
@@ -366,7 +366,7 @@ protected function validate(&$file)
         $types = $this->getAcceptTypes();
 
         //TODO check mime type.
-        if (!in_array(Tools::strtolower(pathinfo($file['name'], PATHINFO_EXTENSION)), $types)) {
+        if (!empty($types) && !in_array(Tools::strtolower(pathinfo($file['name'], PATHINFO_EXTENSION)), $types)) {
```
Comment (PierreRambaud) [inline classes/Uploader.php:369]:
```suggestion
        if (is_array($types) && !in_array(Tools::strtolower(pathinfo($file['name'], PATHINFO_EXTENSION)), $types)) {
```

File: src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php [hunk 3]
```diff
@@ -307,6 +308,11 @@ public function editAction($categoryId, Request $request)
 
         $defaultGroups = $this->get('prestashop.adapter.group.provider.default_groups_provider')->getGroups();
 
+        // If we don't create the dispatcher instance with the current request,
+        // a new instance will be created later using `SymfonyRequest::createFromGlobals()`
+        // but as we may have already uploaded files, this can throw an exception
+        Dispatcher::getInstance($request);
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php:316]: :vomiting_face:  Sorry, bad digestion

File: classes/Uploader.php (hunk 2 above)
Comment (atomiix) [reply classes/Uploader.php:369]: An empty array is not valid here, that's why I use `empty`

File: src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php (hunk 3 above)
Comment (atomiix) [reply src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php:316]: I know 😭

**Comment by atomiix** [comment]
@marwachelly I updated my PR, a better error message is displayed when uploading pdf files. For the delete picture issue, this is solved in [another PR](https://github.com/PrestaShop/PrestaShop/pull/27228). Maybe you can validate both at the same time or just don't validate the delete action in this PR as you'll have to do it in the other?

File: classes/Uploader.php (hunk 2 above)
Comment (PierreRambaud) [reply classes/Uploader.php:369]: I miss read,my bad :+1:

**Review by PierreRambaud — APPROVED** [review]

**Review by saulaski — APPROVED** [review]

**Review by Julievrz — APPROVED** [review]
Wording review ✅

**Review by marwachelly — APPROVED** [review]
Hello, 

PR is tested and the issues https://github.com/PrestaShop/PrestaShop/issues/27237 & https://github.com/PrestaShop/PrestaShop/issues/27303 are fixed.
https://watch.screencastify.com/v/UYAgSW02lfGo12U40nUQ

@HanaRebaiQA Could you please check this [PR](https://github.com/PrestaShop/PrestaShop/pull/27228)

Many thanks,

**Comment by Progi1984** [comment]
Thanks @atomiix & @marwachelly

**Pull request merged**